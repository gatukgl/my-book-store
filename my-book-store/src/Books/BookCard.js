import { Link } from 'react-router-dom'

export const BookCard = ({ book }) => (
  <div className="col mb-5" key={book.id} role="bookCard">
    <div className="card h-100">
      <img className="card-img-top" src={book.coverImage} alt={book.bookName} />

      <div className="card-body p-4">
        <div className="text-center">
          <h5 className="fw-bolder">{book.bookName}</h5>
          <span>${book.price}</span>
        </div>
      </div>
      <div className="card-footer p-4 pt-0 border-top-0 bg-transparent">
        <div className="text-center">
          <Link
            className="btn btn-outline-dark mt-auto"
            to={`/${book.id}`}
            role="moreInfo"
          >
            More Info
          </Link>
        </div>
      </div>
    </div>
  </div>
)
