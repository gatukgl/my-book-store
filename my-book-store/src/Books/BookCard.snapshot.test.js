import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

import { BookCard } from './BookCard'

describe('Book Card Snapshot', () => {
  test('should render correctly', () => {
    const book = {}

    const component = renderer.create(
      <BrowserRouter>
        <BookCard book={book} />
      </BrowserRouter>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should render correctly when has book info', () => {
    const book = {
      coverImage: 'http://image.url',
      bookName: 'Fake book name',
      price: 3.0,
      id: 0,
    }

    const component = renderer.create(
      <BrowserRouter>
        <BookCard book={book} />
      </BrowserRouter>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
