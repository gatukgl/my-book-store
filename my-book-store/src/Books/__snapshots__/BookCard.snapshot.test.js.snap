// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Card Snapshot should render correctly 1`] = `
<div
  className="col mb-5"
  role="bookCard"
>
  <div
    className="card h-100"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body p-4"
    >
      <div
        className="text-center"
      >
        <h5
          className="fw-bolder"
        />
        <span>
          $
        </span>
      </div>
    </div>
    <div
      className="card-footer p-4 pt-0 border-top-0 bg-transparent"
    >
      <div
        className="text-center"
      >
        <a
          className="btn btn-outline-dark mt-auto"
          href="/undefined"
          onClick={[Function]}
          role="moreInfo"
        >
          More Info
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Book Card Snapshot should render correctly when has book info 1`] = `
<div
  className="col mb-5"
  role="bookCard"
>
  <div
    className="card h-100"
  >
    <img
      alt="Fake book name"
      className="card-img-top"
      src="http://image.url"
    />
    <div
      className="card-body p-4"
    >
      <div
        className="text-center"
      >
        <h5
          className="fw-bolder"
        >
          Fake book name
        </h5>
        <span>
          $
          3
        </span>
      </div>
    </div>
    <div
      className="card-footer p-4 pt-0 border-top-0 bg-transparent"
    >
      <div
        className="text-center"
      >
        <a
          className="btn btn-outline-dark mt-auto"
          href="/0"
          onClick={[Function]}
          role="moreInfo"
        >
          More Info
        </a>
      </div>
    </div>
  </div>
</div>
`;
